src/mappings/token.ts

```diff
@@ -62,67 +62,59 @@ function deriveStellarPublicKeyFromHex(issuer: string) {
     return StrKey.encodeEd25519PublicKey(buffer)
 }
 
-function beautifyCurrencyIdString(event: any) {
-    let currencyId = ''
-
-    switch (event.currencyId.__kind) {
+export function beautifyCurrencyIdString(currencyId: any) {
+    switch (currencyId.__kind) {
```

Comment:
We save a line of code here but I think it is not super clean (rather unexpected and surprising) to overwrite the argument variable and to even overwrite it in a way that its type changes.

This makes the code in this function brittle as it is not obvious where the variable still has its old value and where it already has its new value.

Instead: replace the `break` statements by returning the result value directly in each case.